istioctl/pkg/waypoint/waypoint.go

```diff
@@ -414,3 +428,19 @@ func annotateNamespaceWithWaypoint(kubeClient kube.CLIClient, ns string) error {
 	}
 	return nil
 }
+
+func namespaceIsLabeledAmbient(kubeClient kube.CLIClient, ns string) (bool, error) {
+	nsObj, err := kubeClient.Kube().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return false, fmt.Errorf("namespace: %s not found", ns)
+	} else if err != nil {
+		return false, fmt.Errorf("failed to get namespace %s: %v", ns, err)
+	}
+	if nsObj.Labels == nil {
+		return false, nil
+	}
+	if nsObj.Labels[constants.DataplaneMode] == constants.DataplaneModeAmbient {
+		return true, nil
+	}
+	return false, nil
```

Comment:
possible change: not sure this actually reads any more clearly
```suggestion
	return nsObj.Labels[constants.DataplaneMode] == constants.DataplaneModeAmbient, nil
```

Reply:
This is cleaner tbh. Will update